lessons/modelevaluation.ipynb

```diff
@@ -0,0 +1,786 @@
+    "\n",
+    "For each of the priors on $\\theta$ below, (a) sketch the likelihood and prior as a function of theta, (b) calculate the evidence (well enough to comment about the relative differences, anyway).\n",
+    "1. $P(\\theta)$ uniform on $[-1,+1]$\n",
+    "2. $P(\\theta)$ uniform on $[-3,+3]$\n",
+    "3. $P(\\theta)$ uniform on $[-100,+100]$\n",
+    "4. $P(\\theta)$ uniform on $[-\\infty,+\\infty]$\n",
+    "5. $P(\\theta)$ uniform on $[+3,+5]$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "**Evidence exercise: notes**\n",
+    "\n",
+    "1) Models that are less prescriptive (in terms of their priors) are penalized in the evidence. This is a feature, although it means we need to put real thought into those priors.\n",
+    "\n",
+    "2) The evidence can be made arbitrarily small by increasing the prior volume: comparing evidences is more conservative than focusing on the goodness of fit ($L_{\\rm max}$) alone. \n",
+    "\n",
+    "3) The evidence is linearly sensitive to prior volume ($f$), but exponentially sensitive to goodness of fit ($L_{\\rm max} \\propto e^{-\\chi^2_{\\rm min}/2}$). It's still a likelihood, after all."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "The evidence for model $H$, $P(D\\,|\\,H)$, enables a form of Bayesian hypothesis testing: model comparison with the \"evidence ratio\" or \"odds ratio\" or \"Bayes Factor\" $R$\n",
+    "\n",
+    "$R = \\frac{P(D|H_2)}{P(D|H_1)}$\n",
+    "\n",
+    "$R$ is a *fully marginalized likelihood ratio* - which is to say that it *takes into account our uncertainty about values of the parameters of each model by integrating over all plausible values of them.*"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Notice that if your two models are equally probable _a priori_, then \n",
+    "\n",
+    "$\\frac{P(H_2)}{P(H_1} = 1$ such that $\\frac{P(H_2|D)}{P(H_1|D)} = R$\n",
+    "\n",
+    "This assumption is often not always easy to justify."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "A more practical way to interpret the Bayes factor is to note that it updates the model prior ratio into a posterior one. This means that:\n",
+    "\n",
+    "  * If you think that, despite having seen the data, your two models are *still equally probable,*\n",
+    "  \n",
+    "  * then the evidence ratio in favor of $H_2$ is _the odds that you would have had to have been willing to take against $H_2$, before seeing the data._"
```

Comment:
I'm not sure the weirdness of this a posteriori reasoning comes across when just reading the notes

Reply:
Attempted to clarify. And will not emphasize in class...